src/Form/ForecastReportType.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Form;
+
+use App\Entity\DataProvider;
+use App\Model\Reports\ForecastReportFormData;
+use App\Repository\DataProviderRepository;
+use App\Service\ForecastReportService;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\DateType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ForecastReportType extends AbstractType
+{
+    public function __construct(
+        private readonly DataProviderRepository $dataProviderRepository,
+        private readonly ?string $defaultDataProvider,
+        private readonly ForecastReportService $forecastReportService,
+    ) {
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $dataProviders = $this->dataProviderRepository->findAll();
```

Comment:
Why do we need data providers?
Shouldn't we just show data from all or is there a need for this filter?

Reply:
It doesn't make sense to keep. I removed it.